pkg/startup/startup.go

```diff
@@ -282,17 +274,26 @@ func determineNodeName() string {
 	return nodeName
 }
 
-// nodenameFromFile reads the /var/lib/calico/nodename file if it exists and
+func nodenameFileName() string {
+	fn := os.Getenv("CALICO_NODENAME_FILE")
```

Comment:
Do we really want this to be configurable?

What about we have a `nodenameFile` defined in `startup_linux.go` which simply returns `/var/lib/calico/nodename` always?

I think setting CALICO_NODENAME_FILE to something else right now will break the CNI plugin anyway.